subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuild.java

```diff
@@ -42,8 +42,6 @@
     private DefaultDependencySubstitutions dependencySubstitutions;
 
     private GradleLauncher gradleLauncher;
-    private SettingsInternal settings;
-    private GradleInternal gradle;
```

Comment:
These changes got me to thinking that we should make the `IncludedBuildControllers` deal with `BuildController`s only. (or `IncludedBuildController` that wraps/extends a `BuildController`).  And the `DefaultIncludedBuild` should then just be a dumb container for the declared substitutions, name and project path. 

I think that would mean whenever we evaluate the root settings.gradle to build the list of included builds, we should go ahead and configure the settings of those included builds, so this will be simpler (and not reaching into a build controller to get the settings itself).